frame/executive/src/lib.rs

```diff
@@ -597,6 +620,18 @@ mod tests {
 					Call::allowed_unsigned(..) => Ok(Default::default()),
 					_ => UnknownTransaction::NoUnsignedValidator.into(),
 				}
+
+			}
+
+			// Inherent call is accepted for being dispatched
+			fn pre_dispatch(
+				call: &Self::Call,
+			) -> Result<(), TransactionValidityError> {
+				match call {
+					Call::allowed_unsigned(..) => Ok(()),
+					Call::inherent_call(..) => Ok(()),
```

Comment:
I don't know if this is a valid implementation, does any transaction goes through validate_unsigned before getting included in the block ? if so maybe we should allow inherent to be send from local and in_block source.
But then ppl should be careful in offchain worker not to send those inherent transaction or it will produce invalid block.

Reply:
I think the implementation is ok, pre_dispatch allows inherent, validate_unsigned must prevent any non trusted source to send an inherent.